### PR TITLE
Fix 18210

### PR DESCRIPTION
**Pull request by alex-code**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #18210

Allow strict comparison for multi select inputs.
`in_array` does loose checking so can have unexpected results.

```php
$a = ['1.1', '1.10'];

foreach ($a as $v) {
    var_dump(in_array($v, ['1.1']));
}
// Returns, `true`, `true`.
foreach ($a as $v) {
    var_dump(in_array($v, ['1.1'], true));
}
// Returns, `true`, `false`.
```

### CODE REVIEW

File: framework/helpers/BaseHtml.php [hunk 1]
```diff
@@ -819,6 +819,7 @@ protected static function booleanInput($type, $name, $checked = false, $options
      *   Defaults to false.
      * - encode: bool, whether to encode option prompt and option value characters.
      *   Defaults to `true`. This option is available since 2.0.3.
+     * - strict: boolean, if `$selection` is an array and this value is true a strict comparison will be performed on `$items` keys. Defaults to false.
```
Comment (samdark) [inline framework/helpers/BaseHtml.php:822]: Need to add "since 2.0.37"

File: framework/helpers/BaseHtml.php [hunk 2]
```diff
@@ -931,6 +932,7 @@ public static function listBox($name, $selection = null, $items = [], $options =
      *   This option is available since version 2.0.16.
      * - encode: boolean, whether to HTML-encode the checkbox labels. Defaults to true.
      *   This option is ignored if `item` option is set.
+     * - strict: boolean, if `$selection` is an array and this value is true a strict comparison will be performed on `$items` keys. Defaults to false.
```
Comment (samdark) [inline framework/helpers/BaseHtml.php:935]: Need to add "since 2.0.37"

**Review by samdark — APPROVED** [review]

File: framework/helpers/BaseHtml.php (hunk 1 above)
Comment (alex-code) [reply framework/helpers/BaseHtml.php:822]: I don't think `booleanInput` is effected by this.

Comment (alex-code) [reply framework/helpers/BaseHtml.php:822]: Never mind, it's showing above as `booleanInput` but I've added since to the `dropDownList` method.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thank you!

**Pull request merged**